php/class-gsuite-provision.php

```diff
@@ -0,0 +1,196 @@
+<?php
+
+class GSuite_Provision {
+
+	/**
+	 * @var object
+	 */
+	private static $instance;
+
+	/**
+	 * Instance generator.
+	 *
+	 * @return object
+	 */
+	public static function instance() {
+		if ( ! isset( self::$instance ) ) {
+			self::$instance = new GSuite_Provision;
+			self::setup();
+		}
+		return self::$instance;
+	}
+
+	/**
+	 * Adds relevant action and filter hooks.
+	 */
+	private static function setup() {
+		if (
+			! get_option( 'gsuite_domain' ) ||
+			! get_option( 'gsuite_client_id' ) ||
+			! get_option( 'gsuite_client_secret' ) ||
+			// This is a security firewall, since setting this to 'gmail.com' would grant automatic access to the public.
+			'gmail.com' === get_option( 'gsuite_domain' )
+		) {
+			return;
+		}
+
+		add_action( 'login_form', [ self::$instance, 'login_form' ] );
+		add_action( 'login_head', [ self::$instance, 'login_head' ] );
+		add_action( 'login_footer', [ self::$instance, 'login_footer' ] );
+		add_action( 'login_form_gsuite', [ self::$instance, 'login_form_gsuite' ] );
+
+		add_filter( 'login_message', [ self::$instance, 'login_message' ] );
+	}
+
+	/**
+	 * Adds JS to the footer to hide default elements of the login form.
+	 */
+	public function login_footer() {
+		if ( empty( $_GET['gsuite'] ) || 'disable' !== $_GET['gsuite'] ) {
+		?>
+			<script>
+				document.querySelector('#loginform').action = '<?php echo esc_js( GSUITE_PROVISION_URL . '/lib/auth.php' ); ?>'
+				var items = document.querySelector('#loginform').querySelectorAll('p:not(.gsuite)');
```

Comment:
Since you are performing multiple actions on `#loginform` it would be better to cache the lookup:

```javascript
var loginForm = document.getElementById('loginform');
```